tslearn/tests/test_serialize_models.py

```diff
@@ -0,0 +1,178 @@
+import os
+from glob import glob
+import numpy
+import pytest
+from sklearn.exceptions import NotFittedError
+from tslearn import hdftools
+from tslearn.clustering import KShape, TimeSeriesKMeans, GlobalAlignmentKernelKMeans
+from tslearn.metrics import sigma_gak
+from tslearn.datasets import CachedDatasets
+from tslearn.preprocessing import TimeSeriesScalerMeanVariance, TimeSeriesResampler
+
+
+tmp_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'tmp')
+all_formats = ['json', 'hdf5', 'pickle']
+
+try:
+    os.makedirs(tmp_dir)
+except (FileExistsError, OSError):
+    pass
+
+
+def teardown_module():
+    clear_tmp()
+    os.removedirs(tmp_dir)
+
+
+def clear_tmp():
+    files = glob(os.path.join(tmp_dir, '*'))
+    for f in files:
+        os.remove(f)
+
+
+def test_hdftools():
+    dtypes = [numpy.int, numpy.int8, numpy.int16, numpy.int32, numpy.int64,
+              numpy.float, numpy.float32, numpy.float64]
+
+    d = {}
+
+    for dtype in dtypes:
+        name = numpy.dtype(dtype).name
+        d[name] = (numpy.random.rand(100, 100) * 10).astype(dtype)
+
+    fname = os.path.join(tmp_dir, 'hdf_test.hdf5')
+
+    hdftools.save_dict(d, filename=fname, group='data')
+
+    d2 = hdftools.load_dict(fname, 'data')
+
+    for k in d2.keys():
+        numpy.testing.assert_equal(d[k], d2[k])
+
+
+def _check_not_fitted(model):
+    # not serializable if not fitted
+    with pytest.raises(NotFittedError):
```

Comment:
Here, shouldn't the `with` block be located inside the loop such that we ensure all formats would raise an exception if the model is not fitted?